api2-war/src/main/java/eu/europeana/api2/v2/web/controller/SearchController.java

```diff
@@ -177,7 +177,13 @@ public ModelAndView searchJsonGet(
             HttpServletRequest request,
             HttpServletResponse response) throws EuropeanaException {
 
-        LimitResponse limitResponse = apiKeyUtils.checkLimit(wskey, request.getRequestURL().toString(), RecordType.SEARCH, profile);
+        // EA-1826
+        LimitResponse limitResponse;
+        if (StringUtils.equalsIgnoreCase(urlService.getApikeyValidateUrl(), NOTHING)){
+            limitResponse = apiKeyUtils.checkLimit(wskey);
+        } else {
+            limitResponse = apiKeyUtils.validateApiKey(wskey);
+        }
```

Comment:
Something like this should be part of the ApiKeyUtils, so ObjectController and QueryTranslationController can use it as well.

Reply:
Indeed.
It's been moved to ApiKeyUtils.